job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql

```diff
@@ -36,13 +36,14 @@ BEGIN
         WITH RECURSIVE all_job_dependencies AS (
             SELECT partition_id, dependent_job_id, job_id
             FROM job_dependency
-            UNION ALL
+            WHERE partition_id = in_partition_id AND job_id = in_job_id
+            UNION
             SELECT adj.partition_id, adj.dependent_job_id, jd.job_id
             FROM all_job_dependencies adj
             INNER JOIN job_dependency jd ON adj.partition_id = jd.partition_id AND adj.job_id = jd.dependent_job_id
+            WHERE (jd.job_id <> '') IS NOT TRUE
```

Comment:
Is this just the same as `WHERE jd.job_id = ''`?  Why are we doing this?

Reply:
I meant to write the equivalent of `WHERE jd.job_id IS NOT NULL AND jd.job_id <> ''` and ended up flipping the meaning by mistake. This obviously meant the iterative part of the query never returned results and I mistakenly believed I'd solved the performance issue as the query plan/stats where suddenly much better.